e2e/testsuite/testsuite.go

```diff
@@ -374,3 +379,38 @@ func GetNativeChainBalance(ctx context.Context, chain ibc.Chain, user *ibc.Walle
 	}
 	return bal, nil
 }
+
+// ExecuteGovProposal submits the given governance proposal using the provided user and uses all validators to vote yes on the proposal.
+// It ensure the proposal successfully passes.
+func (s *E2ETestSuite) ExecuteGovProposal(ctx context.Context, chain *cosmos.CosmosChain, user *ibc.Wallet, content govtypes.Content) {
+	sender, err := sdk.AccAddressFromBech32(user.Bech32Address(chain.Config().Bech32Prefix))
+	s.Require().NoError(err)
+
+	msgSubmitProposal, err := govtypes.NewMsgSubmitProposal(content, sdk.NewCoins(sdk.NewCoin(chain.Config().Denom, govtypes.DefaultMinDepositTokens)), sender)
+	s.Require().NoError(err)
+
+	txResp, err := s.BroadcastMessages(ctx, chain, user, msgSubmitProposal)
+	s.Require().NoError(err)
+	s.AssertValidTxResponse(txResp)
+
+	// TODO: replace with parsed proposal ID from MsgSubmitProposalResponse
+	// https://github.com/cosmos/ibc-go/issues/2122
+
+	proposal, err := s.QueryProposal(ctx, chain, 1)
+	s.Require().NoError(err)
+	s.Require().Equal(govtypes.StatusVotingPeriod, proposal.Status)
+
+	err = chain.VoteOnProposalAllValidators(ctx, "1", ibc.ProposalVoteYes)
+	s.Require().NoError(err)
+
+	// ensure voting period has not passed before validators finished voting
+	proposal, err = s.QueryProposal(ctx, chain, 1)
+	s.Require().NoError(err)
+	s.Require().Equal(govtypes.StatusVotingPeriod, proposal.Status)
+
+	time.Sleep(time.Second * 30) // pass proposal
```

Comment:
should we maybe wait slightly longer the specified time, maybe we could run into an issue where it is not yet passed.

Or better yet we could wait for the status to equal passed with a timeout of a little over 30 seconds. What do you think?

Reply:
I like your suggestion [here](https://github.com/cosmos/ibc-go/issues/1981#issuecomment-1220482819). Maybe we could fix this in a followup pr? 

I can modify the time to be slightly longer than the voting period, but I figure we will fine tune this functionality over time anyways.

Reply:
yep that sounds good to be 👍